### PR TITLE
Added priority class for authorization server of reversed vpn.

### PR DESCRIPTION
**Pull request by ScheererJ**
Co-authored-by: Sebastian Stauch <sebastian.stauch@sap.com>

**How to categorize this PR?**

/area networking
/kind bug

**What this PR does / why we need it**:
Added priority class for authorization server of reversed vpn.
In some environments, most notably GKE, priority class `system-cluster-critical` is handled in a special way requiring a `ResourceQuota` for usage. Our options were to either create a specific `ResourceQuota` for the `garden` namespace or use a different priority class. This pull request uses the latter approach to resolve the issue.

**Which issue(s) this PR fixes**:
None.

**Special notes for your reviewer**:

**Release note**:

```other operator
None
```

### CODE REVIEW

File: pkg/operation/botanist/component/extauthzserver/external_authz_server.go [hunk 1]
```diff
@@ -127,7 +138,7 @@ func (a *authServer) Deploy(ctx context.Context) error {
 						},
 					},
 					AutomountServiceAccountToken: pointer.Bool(false),
-					PriorityClassName:            "system-cluster-critical",
+					PriorityClassName:            DeploymentName,
```
Comment (rfranzke) [inline pkg/operation/botanist/component/extauthzserver/external_authz_server.go]:
```suggestion
					PriorityClassName:            pc.Name,
```

**Review by ialidzhikov — CHANGES_REQUESTED** [review]
Can you add a proper explanation in the PR description why exactly we need this custom PriorityClass and why it was not possible to use `system-cluster-critical` in some environments?

File: pkg/operation/botanist/component/extauthzserver/external_authz_server.go (hunk 1 above)
Comment (DockToFuture) [reply pkg/operation/botanist/component/extauthzserver/external_authz_server.go]: Updated it.

**Review by rfranzke — APPROVED** [review]
/lgtm

**Comment by ScheererJ** [comment]
> Can you add a proper explanation in the PR description why exactly we need this custom PriorityClass and why it was not possible to use `system-cluster-critical` in some environments?

@ialidzhikov I added a description to the pull request. Does it fulfil your requirements?

**Comment by ialidzhikov** [comment]
> @ialidzhikov I added a description to the pull request. Does it fulfil your requirements?

Thank you!

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**